### PR TITLE
(alan) Added forking mechanism for msg validator 

### PR DESCRIPTION
**Pull request by AKorpusenko**

### CODE REVIEW

**Comment by y0sher** [comment]
https://github.com/ssvlabs/ssv/pull/1486

File: cli/operator/node.go [hunk 1]
```diff
@@ -219,26 +219,21 @@ var StartNodeCmd = &cobra.Command{
...
-		} else {
-			messageValidator = genesisvalidation.New(
+		messageValidator := validation.New(
+			networkConfig,
+			validatorStore,
+			dutyStore,
+			signatureVerifier,
+			validation.WithLogger(logger),
+			validation.WithMetrics(metricsReporter),
+			validation.WithGenesisValidator(genesisvalidation.New(
```
Comment (nkryuchkov) [inline cli/operator/node.go:229]: As a small optimization, we can check if the fork happened and don't even create the genesis message validator in this case

File: networkconfig/config.go [hunk 2]
```diff
@@ -77,6 +77,5 @@ func (n NetworkConfig) GetGenesisTime() time.Time {
 }
 
 func (n NetworkConfig) AlanFork() bool {
-	// TODO: implement
-	return true
+	return n.Beacon.EstimatedCurrentEpoch() >= n.AlanForkEpoch
 }
```
Comment (y0sher) [inline networkconfig/config.go:81]: we should have this or the other not both.

**Comment by y0sher** [comment]
I honestly like [this](https://github.com/ssvlabs/ssv/pull/1486) approach better because it doesn't change the new message validation code.
It's really a matter of preference I guess but maybe @nkryuchkov, @moshe-blox and @MatusKysel  can chime to finalize this.

File: networkconfig/config.go (hunk 2 above)
Comment (AKorpusenko) [reply networkconfig/config.go:81]: Agree. I was not sure does it matter to provide slot as param or not for `AlanForked(slot)`

File: cli/operator/node.go (hunk 1 above)
Comment (AKorpusenko) [reply cli/operator/node.go:229]: Agree. Maybe it even worths to do it inside the `WithGenesisValidator` option.

**Comment by nkryuchkov** [comment]
@y0sher I agree with both approaches but since this PR is closed I finished your PR https://github.com/ssvlabs/ssv/pull/1486

**Pull request closed without merging**